gson/src/main/java/com/google/gson/FieldNamingStrategy.java

```diff
@@ -33,8 +35,18 @@ public interface FieldNamingStrategy {
    * Translates the field name into its JSON field name representation.
    *
    * @param f the field object that we are translating
-   * @return the translated field name.
+   * @return the list of possible translated field names.
    * @since 1.3
    */
   public String translateName(Field f);
+
+  /**
+   * Translates the field name into its JSON field alternative names representation.
```

Comment:
Would probably be good to mention that this only affects deserialization, and maybe also mention `@SerializedName#alternate`, for example something like this:
```suggestion
   * Translates the field name into its JSON field alternative names representation,
   * used for deserialization only. This is similar to {@link SerializedName#alternate()}.
```